test/e2e/Tests/Tests/TerminateOrchestratorTests.cs

```diff
@@ -86,18 +84,22 @@ public async Task TerminateTerminatedOrchestration_ShouldFail()
         Assert.NotNull(terminateAgainResponseMessage);
 
         Assert.Contains("StatusCode=\"FailedPrecondition\"", terminateAgainResponseMessage);
-        Assert.Contains($"InvalidOperationException: Cannot terminate the orchestration instance {instanceId} because instance is in the Terminated state.", terminateAgainResponseMessage);
+        Assert.Contains(fixture.functionLanguageLocalizer!.GetLocalizedStringValue("TerminateTerminatedInstance.FailureMessage", instanceId), terminateAgainResponseMessage);
 
         // Give some time for Core Tools to write logs out
         Thread.Sleep(500);
 
-        Assert.Contains(this.fixture.TestLogs.CoreToolsLogs, x => x.Contains("Cannot terminate orchestration instance in the Terminated state.") &&
+        // For some reason, PowerShell does not log these warnings - instead the status code is 410 (Gone) with no log
+        // when the instance is completed
+        if (fixture.functionLanguageLocalizer.GetLanguageType() == LanguageType.DotnetIsolated)
```

Comment:
I'm curious whether this is specific to .NET Isolated or whether it's specific to PowerShell. Given that the comment calls out PowerShell specifically, should we change the if-check to compare against PowerShell instead of .NET Isolated? For example:
```suggestion
        if (fixture.functionLanguageLocalizer.GetLanguageType() != LanguageType.PowerShell)
```

When we add support for more languages (e.g. Python), then we'll know for sure when these checks run, and we can update both the code and the if-check at the same time, as appropriate.

Reply:
Yes - I suspect the same will be true for the other languages, but given the comment language and the fact that we _want_ to know this for sure, I have inverted the check as suggested. This will cause the test to fail on a naive Python implementation if it is not logged, and we can change this code again then if that happens.